### PR TITLE
Add initial multiple choice component

### PR DESCRIPTION
**Pull request by teneightfive**
Adds a new component for multiple choice fields (radio/checkbox) and
support for testing this new method of component.

screen shot 2017-05-22 at 14 42 17

### CODE REVIEW

File: test/components/form-multilpe-select.test.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
+const expectComponent = require('../component-helper').expectComponent
+
+describe('Form multilpe choice component', () => {
```
Comment (jim68000) [inline test/components/form-multilpe-select.test.js]: typo - multiple

File: src/views/_components/form-multiple-choice.njk [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+<fieldset class="form-group{{ ' inline' if inline }}{{ ' error' if error }}">
+  <legend class="form-label-bold" id="group-{{ name }}">
+    {{ label }} {% if optional %}(optional){% endif %}
+    {% if hint %}
+      <span class="form-hint" id="{{ hintId }}">{{ hint }}</span>
+    {% endif %}
+  </legend>
+
+  <div>
```
Comment (tyom) [inline src/views/_components/form-multiple-choice.njk]: Is it needed?

File: src/views/_components/form-multiple-choice.njk [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ #       value: '2'
+ #     }]
+ #   } %}
+ #}
+
+{% set hintId = 'hint-' + name %}
+{% set variant = variant | default('radio') %}
+
+<fieldset class="form-group{{ ' inline' if inline }}{{ ' error' if error }}">
+  <legend class="form-label-bold" id="group-{{ name }}">
```
Comment (tyom) [inline src/views/_components/form-multiple-choice.njk]: Should it be optional? If not, should we document it as required parameter?

File: src/views/_components/form-multiple-choice.njk [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    {{ label }} {% if optional %}(optional){% endif %}
+    {% if hint %}
+      <span class="form-hint" id="{{ hintId }}">{{ hint }}</span>
+    {% endif %}
+  </legend>
+
+  <div>
+    {% for option in children %}
+      {% set inputValue = option.value if 'value' in option else option.label %}
+      {% set inputId = ['input', name, loop.index] | join('--') %}
```
Comment (tyom) [inline src/views/_components/form-multiple-choice.njk]: Is it unique enough? Should the component custom tag generate a UUID in its context?

File: src/views/_components/form-multiple-choice.njk [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+      {% if variant == 'checkbox' %}
+        {% set isSelected = (value and (inputValue in value)) %}
+      {% else %}
+        {% set isSelected = (inputValue == value) %}
+      {% endif %}
+
+      <label
+        for="{{ inputId }}"
+        class="block-label selection-button-{{ variant }}{{ ' selected' if isSelected }}"
```
Comment (tyom) [inline src/views/_components/form-multiple-choice.njk]: `is-selected`?

Comment (teneightfive) [inline src/views/_components/form-multiple-choice.njk]: It's using the current styles applied by `trade-elements` and `govuk-elements`. We can change this when we move to a more BEM based methodology. Moving this to a component will hopefully make that easier.

File: src/views/_components/form-multiple-choice.njk (hunk 4 above)
Comment (teneightfive) [inline src/views/_components/form-multiple-choice.njk]: Are we likely to have multiple form elements with the same name and the same options on the same page?

File: src/views/_components/form-multiple-choice.njk (hunk 2 above)
Comment (teneightfive) [inline src/views/_components/form-multiple-choice.njk]: Probably not at this stage.

File: src/views/_components/form-multiple-choice.njk (hunk 3 above)
Comment (teneightfive) [inline src/views/_components/form-multiple-choice.njk]: I've used a pattern of documenting when a param is optional and separated those params by a line:

```njk
{##
 # @param {string}       [variant]  My required param
 #
 # @param {string}       [hint]     My optional param
 #}
```

Comment (teneightfive) [inline src/views/_components/form-multiple-choice.njk]: Also not sure that the label for the fieldset should be optional.

Comment (teneightfive) [inline src/views/_components/form-multiple-choice.njk]: Actually, correct JSDoc style is optional params are wrapped in `[]`. So should be this:

```njk
{##
 # @param {string}       variant  My required param
 #
 # @param {string}       [hint]     My optional param
 #}
```

**Pull request merged**